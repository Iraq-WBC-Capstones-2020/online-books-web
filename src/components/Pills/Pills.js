import React, { useState } from 'react';
import {
  MDBContainer,
  MDBTabPane,
  MDBTabContent,
  MDBNav,
  MDBNavItem,
  MDBNavLink,
} from 'mdbreact';

function Pills({ tabs }) {
  const [items, setItems] = useState({
    default: '0',
  });

  const togglePills = (type, tab) => (e) => {
    e.preventDefault();
    if (items[type] !== tab) {
      let _items = { ...items };
      _items[type] = tab;
      setItems(_items);
    }
  };

  return (
    <>
      <div className="py-2 border-t-2 border-b-2">
        <MDBContainer>
          <MDBNav className="nav-pills">
            {tabs.map((tab, index) => (
              <MDBNavItem key={index}>
                <MDBNavLink
                  link
                  to="#"
                  active={items['default'] === index.toString()}
                  onClick={togglePills('default', index.toString())}
                >
                  {tab.title}
                </MDBNavLink>
              </MDBNavItem>
            ))}
          </MDBNav>
        </MDBContainer>
      </div>
      <div className="py-4">
        <MDBContainer>
          <MDBTabContent activeItem={items['default']}>
            {tabs.map((tab, index) => (
              <MDBTabPane tabId={index.toString()} key={index}>
                {tab.content}
              </MDBTabPane>
            ))}
          </MDBTabContent>
        </MDBContainer>
      </div>
    </>
  );
}
export default Pills;
