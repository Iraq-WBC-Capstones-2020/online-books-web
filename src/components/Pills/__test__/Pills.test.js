import React from 'react';
import renderer from 'react-test-renderer';
import Pills from '../Pills';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
test('Pills snapshot', () => {
  const history = createMemoryHistory();
  const tabs = [
    {
      title: 'profilePage.about',
      content: <div>about</div>,
    },
    {
      title: 'profilePage.purchaseHistory',
      content: <div>purchaseHistory</div>,
    },
  ];
  const tree = renderer
    .create(
      <Router history={history}>
        <Pills tabs={tabs} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
