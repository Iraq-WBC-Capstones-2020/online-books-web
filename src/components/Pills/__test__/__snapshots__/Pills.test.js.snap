// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pills snapshot 1`] = `
Array [
  <div
    className="py-2 border-t-2 border-b-2"
  >
    <div
      className="container"
      data-test="container"
    >
      <ul
        className="nav nav-pills"
        data-test="nav"
      >
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            className="nav-link Ripple-parent nav-link Ripple-parent active"
            data-test="nav-link"
            href="/"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            profilePage.about
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            className="nav-link Ripple-parent nav-link Ripple-parent"
            data-test="nav-link"
            href="/"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            profilePage.purchaseHistory
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
      </ul>
    </div>
  </div>,
  <div
    className="bg-gray-200 py-4"
  >
    <div
      className="container"
      data-test="container"
    >
      <div
        className="tab-content"
        data-test="tabContent"
      >
        <div
          className="tab-pane active"
          data-test="tab-pane"
          role="tabpanel"
        >
          <div>
            about
          </div>
        </div>
        <div
          className="tab-pane"
          data-test="tab-pane"
          role="tabpanel"
        >
          <div>
            purchaseHistory
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
